### ISSUE TITLE
Azure DevOps Server packaged with outdated agent and can not use this task

### ISSUE THREAD

**Issue by cliftonz1** [microsoft/PR-Metrics#175]
Currently, Azure DevOps Server is packaged with 2.181.2.zip and has no functionality on updating its agents.
Azure DevOps Server users are unable to use this task if they use the agents packaged with ADO.

**Comment by jzielkeisi** [comment]
We use a hosted internal DevOps Server.  I was able to update the agent by downloading the latest agent and then putting it in C:\ProgramData\Microsoft\Azure DevOps\Agents on the DevOps Server.  Went to the Update agent settings and the server picked it up and installed it. 

https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents?view=azure-devops-2019&tabs=browser#can-i-update-my-v2-agents-that-are-part-of-an-azure-devops-server-pool

**Comment by muiriswoulfe** [comment]
Hi @cliftonz1, thanks for reaching out.

I'm only able to test with the cloud-based version of Azure DevOps, and the agent automatically updates there. I hadn't been aware of the difference with the locally deployed version.

I'm currently working on getting a new release out with some improvements. I hope to get that out this week. I'll try to downgrade the agent as part of this, but I need to ensure the functionality continues to work with the older version (although I expect there won't be problems with that). I'll keep you posted. In the interim, you can use the workaround from @jzielkeisi if that is possible for you.

Also, as I mentioned I'm unable to test on the locally deployed version, so after I get the change out, let me know if you notice any additional issues.

**Comment by cliftonz1** [comment]
@jzielkeisi We have been testing out the same however the 2.190+ agents where giving us issues where the queue time would be near 3 minutes for every stage thus we had to role back.
This is why we are using the default agent that currently comes with ADO until we figure out why that is happening.

Otherwise, we are able to use the new agents.

**Comment by cliftonz1** [comment]
@muiriswoulfe Thank you for looking into this. An option you have to test the older agents is to create an agent on a personal windows computer and attach it to your agent pool to test the task out. 

If you need any more information feel free to ask.

**Comment by muiriswoulfe** [comment]
Thanks @cliftonz1. I've got a PR with the change out now (see link above), which I believe should resolve your issue. Feel free to suggest any alterations you might require.

I've been able to verify that the dependencies on the APIs haven't changed since we were using the older agent version, so I believe just downgrading the minimum agent version is safe here. I've also done some testing that further indicates this. But if, after it goes live early next week, you see any issues, let me know and I'll get a fix out quickly.

**Comment by cliftonz1** [comment]
@muiriswoulfe Thank you, I will make any comments I have on the PR.

After some discussion with Microsoft, the increased staging time to 3 minutes instead of 10 on the agent is expected and in there words "is not a bug".
Thus we have decided to stick with the default agent and not increase our agent version as this severely hinders our CI/CD processes.

I appreciate the assistance on this issue.

**Comment by cliftonz1** [comment]
This issue has been resolved by #183.
Thanks for the help.

**Comment by cliftonz1** [comment]
@muiriswoulfe I made a mistake. I am going to wait till this change is deployed to close this.

When do you plan to deploy this change? 
The sooner the better for me.

**Comment by muiriswoulfe** [comment]
Hi @cliftonz1, I need to get one more PR in before I can release. I then hope to get this out on Monday morning. I'll update the thread when it's released.

**Comment by cliftonz1** [comment]
Thank you!

**Comment by muiriswoulfe** [comment]
Hi @cliftonz1, I've published the new version of the extension now (v1.4.0) and I believe this should resolve the issues for you. Let me know how it goes.

**Comment by cliftonz1** [comment]
@muiriswoulfe  I was able to have the task run but I am unable to get it to connection.

Azure DevOps says our build identity can not access the resources it needs.
I have followed this (documentation)[https://docs.microsoft.com/en-us/azure/devops/pipelines/process/access-tokens?view=azure-devops-2020&tabs=yaml#job-authorization-scope] However, this is still not working.

I am unsure if it follows under this or not.
